sound/soc/sof/intel/hda-loader.c

```diff
@@ -148,8 +148,8 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 				       chip->ipc_ack_mask);
 
 	/* step 5: power down cores that are no longer needed */
-	ret = hda_dsp_core_power_down(sdev, chip->host_managed_cores_mask &
-				      ~(chip->init_core_mask));
+	ret = snd_sof_dsp_core_power_down(sdev, chip->host_managed_cores_mask &
+					  ~(chip->init_core_mask));
```

Comment:
@bardliao now that we're doing this, we dont actually need to set enabled_cores_mask to init_cores in loader.c after FW boot is complete, right?

Reply:
@bardliao I agreed with Ranj. After this enabled_cores_mask should have primary core set(chip->init_core_mask). So we don't need to initialize init_core_mask in snd_sof_run_firmware(). Other than this, all good to go.

Reply:
@ranj063 @fredoh9 Yes, that's why I removed all enabled_cores_mask setting outside snd_sof_dsp_core_power_down/up APIs.